src/spacy/domain.clj

```diff
@@ -132,6 +138,12 @@
           (update ::schedule conj scheduled-session)
           (update ::facts into new-facts)))))
 
+(defn event->ui-representation [event]
+  (-> event
+      (assoc ::next-up (first (::waiting-queue event)))
+      (assoc ::waiting-queue (rest (::waiting-queue event)))
+      (assoc ::available-slots (available-slots event))))
+
```

Comment:
Wondering if this belongs in `spacy.domain` or is rather "view model" logic that should be closer to the UI. Maybe later we can find a fitting place for it.

Reply:
Hmm... yes, I think you are right.